### PR TITLE
added page with list of tasks and some changes in task application

### PR DESCRIPTION
**Pull request by alexandrv1111**

### CODE REVIEW

File: codeArena/task/serializers.py [hunk 1]
```diff
@@ -23,12 +23,10 @@ class Meta:
...
 
 class CreateTaskSerializer(serializers.ModelSerializer):
-    unit_test = serializers.FileField(
-        max_length=500, allow_null=True, allow_empty_file=True)
-
+    
     class Meta:
         model = Task
-        exclude = ['_id', 'created_at', 'updated_at']
+        exclude = ['_id', 'created_at', 'updated_at', 'user_id']
```
Comment (MrSancho42) [inline codeArena/task/serializers.py:29]: Why you was added `user_id` to exclude?

File: codeArena/task/serializers.py [hunk 2]
```diff
@@ -23,12 +23,10 @@ class Meta:
 
 
 class CreateTaskSerializer(serializers.ModelSerializer):
-    unit_test = serializers.FileField(
-        max_length=500, allow_null=True, allow_empty_file=True)
-
```
Comment (MrSancho42) [inline codeArena/task/serializers.py:28]: Why you was deleted this field?

File: codeArena/task/serializers.py [hunk 3]
```diff
@@ -37,7 +35,8 @@ def validate_languages(self, value):
 
         language_list = Language.objects.values_list('name', flat=True)
 
-        if set(value).issubset(set(language_list)):
+        if value in set(language_list): # for models.CharField in DB Mongo
+        # if set(value).issubset(set(language_list)):
```
Comment (MrSancho42) [inline codeArena/task/serializers.py]: Remove this comments

File: codeArena/task/views.py [hunk 4]
```diff
@@ -16,10 +16,13 @@ def get(self, request, format=None):
         serializer.is_valid()
         return Response(serializer.data)
 
+    def delete(self, request, format=None):
+        tasks = Task.objects.all()
+        tasks.delete()
+        return Response ("All tasks was deleted")
```
Comment (MrSancho42) [inline codeArena/task/views.py]: You added delete HTTP method in class what named **Get**TaskListView. Maybe you need to create new view or rename this. And why do we need deleting all task?

Comment (s0fko) [inline codeArena/task/views.py]: Does we need this method? Or is it just for testing?

File: codeArena/task/models.py [hunk 5]
```diff
@@ -29,15 +29,15 @@ class Status(models.TextChoices):
     name = models.CharField(max_length=150, unique=True)
     description = models.TextField()
     user_id = models.UUIDField()
-    unit_test = models.FileField(max_length=500)   # todo: upload_to= ?
+    unit_test = models.TextField()   # todo: upload_to= ?
```
Comment (okslisovska) [inline codeArena/task/models.py]: The comment '#todo: upload_to=' was a reminder and referred to the FileField(). There is no need for this comment anymore.

File: codeArena/task/views.py [hunk 6]
```diff
@@ -16,10 +16,13 @@ def get(self, request, format=None):
...
 
-class GetTaskDetailView(APIView):
-    """Gets object from Task table by _id field"""
```
Comment (s0fko) [inline codeArena/task/views.py]: Was it necessary to delete docstring?)

File: codeArena/task/views.py (hunk 4 above)
Comment (alexandrv1111) [reply codeArena/task/views.py]: just for testing. In the future it could be deleted

File: codeArena/task/views.py [hunk 7]
```diff
@@ -54,8 +78,83 @@ def get(self, request, format='json'):
...
+    def put(self, request, pk):
+        saved_language = Category.objects.filter(_id=ObjectId(pk)).first()
+        data = request.data.get('one_category')
+        serializer = CategorySerializer(
+            instance=saved_language, data=data, partial=True)
+        if serializer.is_valid(raise_exception=True):
+            one_category_saved = serializer.save()
+        return Response({
+            "success": f'The category {one_category_saved.name} updated successfully'
+        })
```
Comment (s0fko) [inline codeArena/task/views.py]: add newline at the end of file pls. Same at other files too

File: front-end/src/Components/pages/tasklistPage/leftcolumn.js [hunk 8]
```diff
@@ -0,0 +1,50 @@
+import React, { Component } from 'react';
+import { Card, Button, Form } from 'react-bootstrap';
+import LanguagesDropDown from '../../Common/LanguagesDropDown'
+import CategoryDropDown from '../../Common/CategoriesDropDown'
+
+export default function Leftcol() {
```
Comment (s0fko) [inline front-end/src/Components/pages/tasklistPage/leftcolumn.js]: Maybe LeftColumn name would be more clear. Same thing with RightColumn in other file

File: codeArena/task/views.py [hunk 9]
```diff
@@ -29,22 +32,43 @@ def get(self, request, pk, format=None):
...
-                    status=status.HTTP_409_CONFLICT)
+                    status=http_status.HTTP_409_CONFLICT)
 
-            return Response(status=status.HTTP_201_CREATED)
+            return Response(status=http_status.HTTP_201_CREATED)
 
-        return Response(status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        # if task.is_valid(raise_exception=True):
+        #     t = task.save()
+            # return Response({"success": f'Task {task.name} created successfully'})
```
Comment (okslisovska) [inline codeArena/task/views.py]: It is not the best practice to keep the commented out code in the repo. 
If there is a reason to keep it in our repo, pls, leave there the comment which explains why.

File: codeArena/task/views.py [hunk 10]
```diff
@@ -29,22 +32,43 @@ def get(self, request, pk, format=None):
...
+    def put(self, request, pk):
+        saved_task = Task.objects.filter(_id=ObjectId(pk)).first()
+        data = request.data.get('one_task')
+        serializer = CreateTaskSerializer(
+            instance=saved_task, data=data, partial=True)
+        if serializer.is_valid(raise_exception=True):
+            one_task_saved = serializer.save()
+        return Response({
+            "success": f'The task {one_task_saved.name} updated successfully'
+        })
```
Comment (MrSancho42) [inline codeArena/task/views.py]: Same problem as highter. This is not logical naming.
Also we not deleting task from table, we only change their status to `inactive` or something else.
And you not return HTTP status, or not using `rest_framework.http_status` for this.

**Review by okslisovska — APPROVED** [review]

File: codeArena/task/views.py [hunk 11]
```diff
@@ -29,22 +32,43 @@ def get(self, request, pk, format=None):
...
+                    status=http_status.HTTP_409_CONFLICT)
 
-            return Response(status=status.HTTP_201_CREATED)
+            return Response(status=http_status.HTTP_201_CREATED)
 
-        return Response(status=status.HTTP_422_UNPROCESSABLE_ENTITY)
+        # if task.is_valid(raise_exception=True):
+        #     t = task.save()
+            # return Response({"success": f'Task {task.name} created successfully'})
+
```
Comment (MrSancho42) [inline codeArena/task/views.py]: Remove this comments

File: codeArena/task/views.py [hunk 12]
```diff
@@ -54,8 +78,83 @@ def get(self, request, format='json'):
...
+    def put(self, request, pk):
+        saved_language = Language.objects.filter(_id=ObjectId(pk)).first()
+        data = request.data.get('one_language')
+        serializer = LanguageSerializer(
+            instance=saved_language, data=data, partial=True)
+        if serializer.is_valid(raise_exception=True):
+            one_language_saved = serializer.save()
+        return Response({
+            "success": f'The language {one_language_saved.name} updated successfully'
+        })
```
Comment (MrSancho42) [inline codeArena/task/views.py:107]: Naming

File: codeArena/task/views.py (hunk 7 above)
Comment (MrSancho42) [inline codeArena/task/views.py]: Also naming

File: front-end/src/Components/pages/tasklistPage/rightcolumn.js [hunk 13]
```diff
@@ -0,0 +1,62 @@
...
+
+                    <Card bg='dark' style={{marginBottom: '10px'}}>
+                        <ListGroup horizontal variant="flush" fluid > 
+                            <ListGroup.Item action href='#task'variant='dark' style={{ height: 60 }}>
+                                {card.name}
+                            </ListGroup.Item>
+                        
+                            <ListGroup.Item variant='dark' style={{ width: '10rem' }}>
+                                {/* <Image src={dict_of_langs[card.languages.substr(2, card.languages.length-4)]} style={{width: '2rem', }}>
+                                </Image> -------if languages will change to array*/}
```
Comment (MrSancho42) [inline front-end/src/Components/pages/tasklistPage/rightcolumn.js]: Comment

File: codeArena/task/serializers.py (hunk 2 above)
Comment (alexandrv1111) [reply codeArena/task/serializers.py:28]: Because unit_test is now TextField

**Pull request closed without merging**